hack/wait-for-buildx.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# File taken from a closed-source project originally written by SgtCoDFish
+# Contributed to the trust-manager project under the above Apache license
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# This script is required as "buildx rm" and "buildx create" can return successfully
+# before they actually finish. We need to ensure that the builder is actually removed
+# or created (as needed) before we proceed.
```

Comment:
note: I didn't want to create this file, but I saw several flakes and failures both locally and in GCB relating to this :cry: